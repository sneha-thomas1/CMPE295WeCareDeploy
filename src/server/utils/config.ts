const config = {};

//module.exports = config;
export default config;
