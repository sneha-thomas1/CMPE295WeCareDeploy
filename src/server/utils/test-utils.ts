export const mockResponse = () => {
    const res: any = {};
    res.status = jest.fn().mockReturnValue(res);
    res.json = jest.fn().mockReturnValue(res);
    res.send = jest.fn((x) => x);
    return res;
};

type optionType = {
    query?: any;
    params?: any;
    body?: any;
    sessionData?: any;
};
export const mockRequest = (options: optionType = {}) => {
    const { query = {}, params = {}, body = {}, sessionData = {}, ...rest } = options;
    return {
        get(name) {
            return jest.fn().mockReturnValue(name);
        },
        session: { data: sessionData },
        header: jest.fn(),
        query,
        body,
        params,
        ...rest,
    };
};
