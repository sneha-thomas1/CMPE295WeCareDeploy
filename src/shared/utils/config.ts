const serverUrl = 'http://localhost:8500/api/';

export default serverUrl;
