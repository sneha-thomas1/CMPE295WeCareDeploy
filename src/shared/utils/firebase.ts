import { initializeApp } from 'firebase/app';
import { getAuth } from 'firebase/auth';
import { getStorage, ref } from 'firebase/storage';
import 'firebase/storage';

const firebaseConfig = {
    // apiKey: process.env.REACT_APP_API_KEY,
    // authDomain: process.env.REACT_APP_AUTH_DOMAIN,
    // projectId: process.env.REACT_APP_PROJECT_ID,
    // storageBucket: process.env.REACT_APP_STORAGE_BUCKET,
    // messagingSenderId: process.env.REACT_APP_MESSAGING_SENDER_ID,
    // appId: process.env.REACT_AP_APP_ID,
    // measurementId: process.env.REACT_APP_MEASUREMENT_ID,
    apiKey: 'AIzaSyBRhY61TgBU4o_iav8XPGUevbIcO74fmgs',
    authDomain: 'cmpe295-wecare.firebaseapp.com',
    projectId: 'cmpe295-wecare',
    storageBucket: 'cmpe295-wecare.appspot.com',
    messagingSenderId: '181804813353',
    appId: '1:181804813353:web:a22ff89a9c9c1397292038',
    measurementId: 'G-JKZYL2KZQV',
};

const app = initializeApp(firebaseConfig);
const storage = getStorage();
const auth = getAuth(app);
export { auth, storage };
