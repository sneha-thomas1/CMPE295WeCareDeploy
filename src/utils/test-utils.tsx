import { Provider } from 'react-redux';
import React from 'react';
import { StaticRouter } from 'react-router-dom';
import { HelmetProvider } from 'react-helmet-async';
import { ThemeProvider, StyledEngineProvider } from '@material-ui/core';
import configureStore from '../shared/store';
import theme from '../shared/theme';

export const withRedux = (Component) => {
    const wrapped: React.FunctionComponent<any> = ({ initialState = {}, store, ...props }) => {
        return (
            <Provider store={store || configureStore({ initialState })}>
                <Component {...props} />
            </Provider>
        );
    };
    wrapped.displayName = Component.displayName || Component.name || 'Component';
    return wrapped;
};
export const withAppProviders = (Component) => {
    const wrapped: React.FunctionComponent<any> = ({ initialState = {}, store, ...props }) => {
        return (
            <Provider store={store || configureStore({ initialState })}>
                <StaticRouter>
                    <HelmetProvider>
                        <StyledEngineProvider injectFirst>
                            <ThemeProvider theme={theme}>
                                <Component {...props} />
                            </ThemeProvider>
                        </StyledEngineProvider>
                    </HelmetProvider>
                </StaticRouter>
            </Provider>
        );
    };
    wrapped.displayName = Component.displayName || Component.name || 'Component';
    return wrapped;
};
