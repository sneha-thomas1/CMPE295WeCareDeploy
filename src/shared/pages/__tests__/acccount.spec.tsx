import React from 'react';
import { cleanup, render, fireEvent } from '@testing-library/react';
import AccountPage from '../Account/Account';
import { withAppProviders } from '../../../utils/test-utils';
const AccountPageWithApp = withAppProviders(AccountPage);
describe('<AccountPage {...defaultProps} />', () => {
    afterEach(cleanup);
    it('should render element', () => {
        const { container } = render(<AccountPageWithApp />);
        expect(container).toMatchSnapshot();
    });
    it('should render text content', () => {
        const { getByText } = render(<AccountPageWithApp />);
        expect(getByText('Email Address')).toBeInTheDocument();
    });
});
