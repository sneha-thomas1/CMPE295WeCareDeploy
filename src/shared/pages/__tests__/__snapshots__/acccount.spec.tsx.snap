// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountPage {...defaultProps} /> should render element 1`] = `
<div>
  <div
    class="MuiBox-root css-2pi2o1"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-8v6pjz-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1x8weih-MuiPaper-root-MuiCard-root"
          >
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <div
                class="MuiBox-root css-1efvhw7"
              >
                <div
                  class="MuiAvatar-root MuiAvatar-circular css-1yhts3d-MuiAvatar-root"
                >
                  <img
                    class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                    src="/static/images/avatars/avatar_2.png"
                  />
                </div>
                <h4
                  class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1ndhd1t-MuiTypography-root"
                />
                <p
                  class="MuiTypography-root MuiTypography-body1 css-ngt9wa-MuiTypography-root"
                >
                  San Jose USA
                </p>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <div
              class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-fullWidth MuiButtonBase-root css-bux2iu-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Upload picture
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 MuiGrid-grid-lg-8 css-1wyz5j-MuiGrid-root"
        >
          <form
            autocomplete="off"
            novalidate=""
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1x8weih-MuiPaper-root-MuiCard-root"
            >
              <div
                class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-15pqjh0-MuiTypography-root"
                  >
                    User Profile
                  </span>
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiCardHeader-subheader css-1atwq5k-MuiTypography-root"
                  >
                    The information can be edited
                  </span>
                </div>
              </div>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
              />
              <div
                class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1tp3ba8-MuiGrid-root"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1y9gz45-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="false"
                      >
                        First name
                        <span
                          aria-hidden="true"
                          class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1uhu8k1-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                          name="firstName"
                          required=""
                          type="text"
                          value=""
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-2s9zix"
                          >
                            <span>
                              First name
                               *
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                      <p
                        class="MuiFormHelperText-root MuiFormHelperText-sizeMedium MuiFormHelperText-contained Mui-required css-19alodt-MuiFormHelperText-root"
                      >
                        Please specify the first name
                      </p>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1tp3ba8-MuiGrid-root"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-1y9gz45-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="false"
                      >
                        Last name
                        <span
                          aria-hidden="true"
                          class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1uhu8k1-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                          name="lastName"
                          required=""
                          type="text"
                          value=""
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-2s9zix"
                          >
                            <span>
                              Last name
                               *
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1tp3ba8-MuiGrid-root"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1wx1fhp-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="true"
                      >
                        Email Address
                        <span
                          aria-hidden="true"
                          class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1uhu8k1-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                          name="email"
                          required=""
                          type="text"
                          value="abc@example.com"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-17xdsu"
                          >
                            <span>
                              Email Address
                               *
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1tp3ba8-MuiGrid-root"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-1y9gz45-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="false"
                      >
                        Phone Number
                      </label>
                      <div
                        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1uhu8k1-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                          name="phone"
                          type="number"
                          value=""
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-2s9zix"
                          >
                            <span>
                              Phone Number
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1tp3ba8-MuiGrid-root"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1wx1fhp-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="true"
                      >
                        Country
                        <span
                          aria-hidden="true"
                          class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1uhu8k1-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                          name="country"
                          required=""
                          type="text"
                          value="USA"
                        />
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-17xdsu"
                          >
                            <span>
                              Country
                               *
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1tp3ba8-MuiGrid-root"
                  >
                    <div
                      class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                    >
                      <label
                        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-required css-1wx1fhp-MuiFormLabel-root-MuiInputLabel-root"
                        data-shrink="true"
                      >
                        Select State
                        <span
                          aria-hidden="true"
                          class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                        >
                           
                          *
                        </span>
                      </label>
                      <div
                        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiSelect-root css-1uhu8k1-MuiInputBase-root-MuiOutlinedInput-root"
                      >
                        <select
                          aria-invalid="false"
                          class="MuiNativeSelect-select MuiNativeSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-85kf4q-MuiNativeSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                          name="state"
                          required=""
                        >
                          <option
                            value="alabama"
                          >
                            Alabama
                          </option>
                          <option
                            value="new-york"
                          >
                            New York
                          </option>
                          <option
                            value="san-francisco"
                          >
                            San Francisco
                          </option>
                        </select>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiNativeSelect-icon MuiNativeSelect-iconOutlined css-1g12qau-MuiSvgIcon-root-MuiNativeSelect-icon"
                          data-testid="ArrowDropDownIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M7 10l5 5 5-5z"
                          />
                        </svg>
                        <fieldset
                          aria-hidden="true"
                          class="MuiOutlinedInput-notchedOutline css-1o2jng6-MuiOutlinedInput-notchedOutline"
                        >
                          <legend
                            class="css-17xdsu"
                          >
                            <span>
                              Select State
                               *
                            </span>
                          </legend>
                        </fieldset>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <hr
                class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
              />
              <div
                class="MuiBox-root css-1qc63lq"
              >
                <button
                  class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-nen8n9-MuiButtonBase-root-MuiButton-root"
                  tabindex="0"
                  type="button"
                >
                  Save details
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
